Feat/video grid

hyuneeekang opened the pull request:
## 개요
- Shorts 컴포넌트화
- 비디오 그리드 내 3번째 행 Shorts Row 고정
- 비디오 썸네일에 영상 시간 뱃지 UI 추가
- 영상 제목 2줄 이상 말줄임표 처리
- padding 및 z-index 등 조정
- PR #10 적용

<!---- Resolves: #(Isuue Number) -->

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [x] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)
<img width="500" alt="스크린샷 2025-04-10 오전 10 32 22" src="https://github.com/user-attachments/assets/037c9b6c-24b1-423b-b124-88f888930a93" />
<img width="500" alt="스크린샷 2025-04-10 오전 10 32 31" src="https://github.com/user-attachments/assets/411fd24c-9be6-4762-9206-ea85ecf215e1" />


## 💬 공유사항 to 리뷰어

- `수정완료` 창 사이즈를 줄이거나 키우면 쇼츠 섹션 사이즈가 좀 이상해 보일텐데요.. 점차 고쳐나가겠습니다...
- `수정완료` 영상 제목 말줄임표 처리했는데, 3번째 줄에 미세하게 글자가 보여요. 이거 어떻게 고치는지 아시는 분 계시면 알려주세요..!
<img width="236" alt="스크린샷 2025-04-10 오전 10 30 59" src="https://github.com/user-attachments/assets/d963922a-1eed-46aa-bd10-8ee5bb56b595" />


## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다. 
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

hyew0 committed:
style: 사이드바 축소 시 컨텐츠 영역이 전체를 차지하도록 수정, 아이콘 색상 변경

hyew0 committed:
fix: main branch 헤더 고정으로 인한 카테고리 영역 미표시 문제 해결

hyuneeekang committed:
style: CSS 수정
- 썸네일 내 영상 시간 UI 추가함
- 영상 제목 2줄 이상 말줄임표 처리함
- padding 및 z-index 등 조정함

hyuneeekang committed:
refactor: Shorts.js 리팩토링
- Shorts Row를 video grid로 export하기 위해 리팩토링 진행
- video grid 내 쇼츠 UI 생성 예정

hyuneeekang committed:
css: grid CSS 편집
- video grid 3번째 행에 shorts 섹션 위치하도록 함

hyuneeekang committed:
Merge commit '3340563e2108287933b7e38e701b7c4c172684b2' into feat/video-grid

Shin-Yu-1 commented on src/style/VideoGrid.css:
```diff
@@ -34,6 +48,14 @@
   font-size: 14px;
   line-height: 20px;
   padding-bottom: 5px;
+ /* 2줄 이상일 때 말줄임표 */
```
max-height를 주면 사라지네여.
유튜브에서 확인해보면 이렇게 있어요!
참고해보시면 좋을 것 같아요! :)

```
color: var(--yt-spec-text-primary);
font-family: "Roboto", "Arial", sans-serif;
font-size: 1.6rem;
line-height: 2.2rem;
font-weight: 500;
overflow: hidden;
/* display: block; */
max-height: 4.4rem;
-webkit-line-clamp: 2;
display: box;
display: -webkit-box;
-webkit-box-orient: vertical;
text-overflow: ellipsis;
white-space: normal;

hyuneeekang replied:
와 해결됐어요! 유튜브에서 확인을 못해봤었네요. 감사합니다!!

sjoon627 commented on src/components/VideoGrid.js:
```diff
@@ -127,19 +128,25 @@ class VideoGrid {
+    this.shortsRow = createElement('div', this.videoGrid, { className: 'shorts-container' });
+    this.shortsRow.innerHTML = Shorts();
+
+    this.calHeight();
```
창 사이즈를 줄이거나 키우면 쇼츠 섹션 사이즈가 의도와 다르게 보이는 문제는
초기화 되었을때만 창 사이즈를 받아와서 생기는 문제같아요

window.addEventListener('resize', this.calHeight); 넣으면 해결될 것 같습니다!

hyuneeekang replied:
오 생각해보니 그렇네요...!!
다른 방법으로 수정하긴 했는데 좋은 의견 감사합니다.

Jammanb0 reviewed (approved):
고생하셨습니다!👍

Jammanb0 reviewed (approved):
정말 고생하셨습니다 👍

hyuneeekang committed:
style: CSS 수정(말줄임표, Shorts 섹션 사이즈)
- 영상 제목 말줄임표 처리 시 줄바꿈된 글자가 조금 보이는 이슈 처리 (CSS)
- 창 사이즈를 줄이거나 키울 때, 쇼츠 섹션 사이즈가 자동으로 조정되지 않는 이슈 처리

yuj118 reviewed (approved):
수고하셨습니다 🥹!!!
